ucsdscheduleme/Repo/ScheduleRepo.cs

```diff
@@ -31,98 +27,105 @@ public class ScheduleRepo
         /// <returns>List of possible schedules for course list.</returns>
         public List<List<Section>> FindScheduleForClasses(Course[] coursesToSchedule)
         {
-            List<Node> AllSections = new List<Node>();
-            List<List<Section>> possibleSchedules = new List<List<Section>>();
+        List<List<Section>> possibleSchedules = new List<List<Section>>();
+
+            // If there is only one class to schedule.
+            if (coursesToSchedule.Count() == 1)
+            {
+                possibleSchedules.Add((System.Collections.Generic.List<ucsdscheduleme.Models.Section>)coursesToSchedule[0].Sections);
+                return possibleSchedules;
+            }
 
             Course firstCourse = coursesToSchedule[0];
             int numClasses = coursesToSchedule.Length;
 
-            // Create a node for each meeting.
-            int numNodes = 0;
-            foreach (Course course in coursesToSchedule)
+            // Create a node for each Section.
+            List<List<Node>> AllSections = new List<List<Node>>();
+
+            for (int i = 0; i < coursesToSchedule.Count(); i++)
             {
-                foreach (Section section in course.Sections)
+                List<Node> thisCourse = new List<Node>();
+                foreach (Section j in coursesToSchedule[i].Sections)
                 {
-                    Node thisSection = new Node(numNodes++)
+                    Node thisSection = new Node()
                     {
-                        Course = course,
-                        Section = section
+                        Course = coursesToSchedule[i],
+                        Section = j,
+                        Level = i
```

Comment:
What is Level1?

Reply:
level is suppose to represent the layer of the tree the node is on